src/queries/makeQueries.ts

```diff
@@ -128,7 +163,13 @@ export function makeQueries<Predicate, Options>(
       }
 
       if (results.length === 0) {
-        throw new ErrorWithStack(getMissingError(predicate, options), getFn);
+        throw new ErrorWithStack(
+          `${getMissingError(
+            predicate,
+            options
+          )}${getWrongHostComponentNamesError()}`,
```

Comment:
I wonder if it would rather be useful to put `getWrongHostComponentNamesError` inside relevant `getMissingError` calls for `*ByText`, etc queries.

Reply:
The reason I did it like this is it looked simpler to do it here because I don't have to call this at several places. It is true that this cannot be related to a failed byTestId query but even if it's not the actual issue users should be aware that their configuration is broken anyway no ? The only downside I see to this approach is running autodetect more often than we need to so from a performance perspective it's not ideal but it only happens on failed tests so it's not very impactful and also it's not very costly

Reply:
The numer of places is limited to 3 calls sites. I think it would make sense from structural POV to run this check only in relevant queries. Especially that it should be also applied to `getAllBy` variant.

‼️ BTW what is the impact on having it run repeatedly in `findBy*` and `findAllBy`?